### PR TITLE
feat(KMultiselect)

### PR DESCRIPTION
**Pull request by travispamaral**
### Summary
Adds new Multiselect component in place of KDropdown. This is roughly modeled after the [Evergreen Select Menu](https://evergreen.segment.com/components/select-menu/). Meant for multiple options (no single only option). Currently no slots & minimal theming.

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

**Comment by travispamaral** [comment]
This is not finished yet, need to work out slots and theming (not sure Im going to add) as well as finish adding in a filter. Still working with design on a few things but figured early review would be nice.

File: docs/components/dropdown.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+      items: [
+        {name: 'item1'},
+        {name: 'item2'},
+        {name:'item3'}
+      ]
+    }
+  }
+}
+</script>
+```
```
Comment (jackkav) [inline docs/components/dropdown.md:45]: I wish there was an automatic way to catch when these don't match, or infer them

**Comment by darrenjennings** [comment]
> Meant for multiple options (no single only option)

This could be implemented by adding a `@selected` event who's function signature takes in both the item that is selected, and all the items e.g. (`@selected="handleClick"` where handleClick emits the event and passes the copied item/items by reference only if the selected listener exists), so that you could optionally deselect all other items - but by default it would only toggle the current item that you clicked.

File: packages/KMultiselect/KMultiselect.vue [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+import KInput from '@kongponents/kinput/KInput.vue'
+
+export default {
+  name: 'KMultiselect',
+  components: { KButton, KPop, KInput },
+  props: {
+    width: {
+      type: String,
+      default: '200'
+
```
Comment (darrenjennings) [inline packages/KMultiselect/KMultiselect.vue:73]:
```suggestion
```

File: docs/components/dropdown.md (hunk 1 above)
Comment (travispamaral) [reply docs/components/dropdown.md:45]: good catch this file should have been removed

**Comment by travispamaral** [comment]
I should have gave a more in depth explanation of what design is looking for in this component. It was decided that this component would always be used for multiple selections and never just one, they decided it would be more beneficial in most cases to use something more like a combobox ([link this](https://evergreen.segment.com/components/combobox)) for a single selection. Additionally it was decided that anywhere this component is used it would use the "apply" button and not have an option to automatically emit selected values out. Because of this decision I did not bake in that functionality. 

@jackkav originally when we talked I said this would be a replacement of that you wound up using in alerts however, the scope of the component changed after discussing with design and it was decided it would not function exactly the same as the `KDropdown` would have.

**Comment by jackkav** [comment]
Your first example has an inconsistency with the Vue code block, "disabled: true"

**Pull request merged**